### PR TITLE
WL-3242

### PR DESCRIPTION
**Pull request by nicholaswilson100**
Add a 'Remove All' checkbox in header
Rename buttons to 'Remove selected polls' and 'Reset selected polls'
Add a JS confirm popup for both buttons

### CODE REVIEW

**Comment by nicholaswilson100** [comment]
I think 'Are you sure you want to remove the selected items?' is a better popup message than 'Are you sure you want to remove the following items?'  so I'll change that

File: tool/src/webapp/templates/votePolls.html [hunk 1]
```diff
@@ -69,15 +82,15 @@ <h3 rsf:id="poll-list-title">Polls</h3>
...
+      <td class="checkbox-cell"><label rsf:id="delete-label" id="check-2" class="skip">Label for element</label><input rsf:id="poll-select" title="Select to remove" type="checkbox" id="check-2" name="deletePoll" value="" />
       </td>
     </tr>
   </tbody>
 </table>
 
 <p class="act">
-<input type="submit" class="active" rsf:id="delete-polls" name="deleteSelectPoll" value="Update"  accesskey="s" />
-<input type="submit" class="active" rsf:id="reset-polls-votes" name="resetSelectPoll" value="Reset"  accesskey="r" />
+<input type="submit" class="active" rsf:id="delete-polls" name="deleteSelectPoll" onclick="return confirm('Are you sure you want to remove the following item(s)?');" value="Update"  accesskey="s" />
```
Comment (buckett) [inline tool/src/webapp/templates/votePolls.html]: The text string should be in an i18n file, I'm not sure how you get at RSF i18n properties in JS though.
It's also normally good practice to separate the JS behaviour from the HTML so I'd attach the onclick using jQuery.

File: tool/src/webapp/templates/votePolls.html [hunk 2]
```diff
@@ -32,6 +32,19 @@ <h3 rsf:id="poll-list-title">Polls</h3>
 	<a href="voteAdd.html" rsf:id="add-poll">Add</a>
 </div>
 
+<script type="text/javascript">
+	$(document).ready(function(){
+	$('#remove-all').click(function(){
+		if ($(this).is(':checked')){
```
Comment (buckett) [inline tool/src/webapp/templates/votePolls.html]: The JavaScript voice in my head likes:
`$('.checkbox-cell input').attr('checked', $(this).is(':checked'));`

**Comment by nicholaswilson100** [comment]
That is nice :)

I think I just pinched some code from somewhere else in WebLearn that does the trick (not great :()).  Ideally, I'd like to separate out the JS from the html file altogether and use Jquery to bind click events (by id ideally as that's less likely to change than classes).
http://api.jquery.com/bind/

**Comment by nicholaswilson100** [comment]
(Of course here there's 2  elements with same class so by class makes more sense.)

**Comment by nicholaswilson100** [comment]
I've included your much nicer JS for the select all checkbox, separated out the JS from the HTML and put the confirm message in a i18n file (using hidden rsf divs).

File: tool/src/java/org/sakaiproject/poll/tool/producers/PollToolProducer.java [hunk 3]
```diff
@@ -185,6 +185,9 @@ public void fillComponents(UIContainer tofill, ViewParameters viewparams,
 		df.setTimeZone(tz);
 		//m_log.debug("got timezone: " + tz.getDisplayName());
 
+        UIOutput.make(tofill, "poll_list_remove_confirm", messageLocator.getMessage("poll_list_remove_confirm"));
```
Comment (buckett) [inline tool/src/java/org/sakaiproject/poll/tool/producers/PollToolProducer.java]: Indentation looks wrong here.

**Comment by nicholaswilson100** [comment]
Fixed indentation

**Comment by nicholaswilson100** [comment]
Be nice if there was a sakai.code.settings.xml (or whatever it is) for different files that anyone could load up for IDEA...

File: tool/src/webapp/templates/votePolls.html [hunk 4]
```diff
@@ -58,7 +77,7 @@ <h3 rsf:id="poll-list-title">Polls</h3>
       <td rsf:id="poll-open-date">opening date</td>
       <td rsf:id="poll-close-date">closing date</td>
       <td><a rsf:id="poll-results">Results</a></td>
-      <td><label rsf:id="delete-label" id="check-2" class="skip">Label for element</label><input rsf:id="poll-select" title="Select to remove" type="checkbox" id="check-2" name="deletePoll" value="" /> 
+      <td class="checkbox-cell"><label rsf:id="delete-label" id="check-2" class="skip">Label for element</label><input rsf:id="poll-select" title="Select to remove" type="checkbox" id="check-2" name="deletePoll" value="" />
```
Comment (buckett) [inline tool/src/webapp/templates/votePolls.html:80]: Duplicate IDs in HTML.
Nesting the input in the label would mean you don't need the for/id attributes.
Sorry for not spotting this first time round.

**Comment by nicholaswilson100** [comment]
It doesn't seem to like nesting the input in the label.  The checkbox disappears.  I think it's because you can't do this when a table is being used for layout;

http://www.w3.org/TR/html401/interact/forms.html#edef-LABEL

**Comment by buckett** [comment]
Ah, ok I think it's disappearing because the label is acutally hidden and is just being used to give the checkbox a tooltip.

**Comment by buckett** [comment]
And actually RSF re-writes the id and for attributes on the elements anyway.

**Pull request merged**